woocommerce/rest-api/Controllers/Settings.php

```diff
@@ -122,7 +122,30 @@ public function get_items_permissions_check( $request ) {
 	}
 
 
-	// TODO: get_items()
+	/**
+	 * Gets all registered settings.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param \WP_REST_Request $request request object
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function get_items( $request ) {
+
+		$items = [];
+
+		foreach ( $this->settings->get_settings() as $setting ) {
+
+			$response = $this->prepare_item_for_response( $setting, $request );
+
+			// check that the response object has data
```

Comment:
Hmmm, maybe we should move `rest_ensure_response` out of the `prepare_item_for_response` at this point? It feels weird to wrap the data array in a response object and then extract the array from it... What do you think?

Reply:
I agree, it feels a bit weird. Since `prepare_item_for_response()` is defined in the parent class, I think we should also rename `prepare_item_for_response()` to, for example, `prepare_setting_item()`, and return the array version there.

What do you think?

Reply:
Sounds good to me. Maybe add a note to the original story documenting the decision.

Reply:
Sounds good. I modified the method in 615ef5d and added a note to the [story](https://github.com/skyverge/wc-plugin-framework/pull/464#discussion_r397551765).